app/repository/pg/conndb.go

```diff
@@ -0,0 +1,91 @@
+package pg
+
+import (
+	"bufio"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+type DB struct {
+	Host     string
+	Port     string
+	Username string
+	Password string
+	DBName   string
+}
+
+func loadEnvVar() error {
+	f, err := os.Open(".env")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			log.Printf("%s", err)
+		}
+	}()
+
+	lines := make([]string, 0)
+	scanner := bufio.NewScanner(f)
+
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	for _, l := range lines {
+		pair := strings.Split(l, "=")
+		if len(pair) != 2 {
+			return errors.New("not enough data for the configuration in .env file")
+		}
+
+		os.Setenv(pair[0], pair[1])
+	}
+
+	return nil
+}
+
+type DBOptions struct {
+	DB DB
+}
+
+func GetConfig() (DBOptions, error) {
```

Comment:
Config should come as an argument to ConnectDB() but it should NOT be defined here.
You have to define it in `cfg` package. 

So,  `DBOptions` and `GetConfig()` should be placed in `cfg` package. This package should know nothing about the way you getting the config.

Reply:
Fixed.